components/director/pkg/handler/timeout_handler.go

```diff
@@ -0,0 +1,38 @@
+package handler
+
+import (
+	"encoding/json"
+	"net/http"
+	"time"
+
+	"github.com/kyma-incubator/compass/components/director/pkg/apperrors"
+)
+
+const (
+	HeaderContentTypeKey   = "Content-Type"
+	HeaderContentTypeValue = "application/json;charset=UTF-8"
+)
+
+func WithTimeout(h http.Handler, timeout time.Duration) (http.Handler, error) {
```

Comment:
logging the error from second writing might be useful, but we can discuss this in the 2nd PR where we have usage examples also